palantir-java-format/src/main/java/com/palantir/javaformat/doc/Level.java

```diff
@@ -296,8 +296,14 @@ private Exploration breakNormally(State state, LevelNode levelNode, CommentsHelp
         if (keepIndent) {
             newState = newState.withIndentIncrementedBy(getPlusIndent());
         }
-        return Optional.of(tryToLayOutLevelOnOneLine(
-                commentsHelper, maxWidth, newState, memoizedSplitsBreaks.get(), explorationNode));
+        State partiallyInlinedState = tryToLayOutLevelOnOneLine(
```

Comment:
why not move this down under the bodyIsComplex calculation so we can short-circuit and avoid computing it if bodyIsComplex == false?